### DISCUSSION TITLE
Consumer buffer size control

### DISCUSSION THREAD

**Discussion by MetalRex101** [rabbitmq/amqp091-go#190] [Q&A]
### Is your feature request related to a problem? Please describe.

Having: A queue with 1 million messages, that needs to be consumed.

Once consumer started, it will read all the available messages into a buffer, which will cause the following:

1. Rabbit MQ will load all the messages from a disk to a RAM;
2. It will drastically increase client application memory consumption.

### Describe the solution you'd like

It would be nice to have a setting for consumer buffer size (per consumer?).
Once buffer size is reached, RMQ client will not read new messages from RabbitMQ.

### Describe alternatives you've considered

Alternative is to set fixed-sized buffer. Everything is better, than unlimited one.

**Comment by Zerpet** [comment]
The [consumer prefetch count](https://rabbitmq.com/confirms.html#channel-qos-prefetch) is the feature you are looking for. This is exposed in `Channel.Qos()` function: https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Channel.Qos

By default there is no limit. You can set a default limit at the server level using the advanced config and the setting described here: https://rabbitmq.com/consumer-prefetch.html#default-limit

**Comment by MetalRex101** [comment]
@Zerpet, First of all, thank you very much for such a fast reply.
I will try it right now!

**Comment by MetalRex101** [comment]
It solved my issue! Thank you!

**Comment by lukebakken** [comment]
@MetalRex101 thanks for using RabbitMQ.

In general, if a GitHub project has "Issues" and "Discussions", it's best to start a new discussion unless you are _certain_ you have found a bug, or that the feature you request isn't already implemented.
